src/wire/mod.rs

```diff
@@ -431,4 +450,31 @@ mod tests {
             .to_string()
             .contains("string length 0 outside of allowed range 1..=128"));
     }
+
+    #[test]
+    fn deserialize_string_within_range_inclusive() {
+        use serde::Deserialize;
+
+        #[derive(Debug, Deserialize, PartialEq, Eq)]
+        struct Test(
+            #[serde(deserialize_with = "super::string_within_range_inclusive::<1, 128, _>")] String,
+        );
+
```

Comment:
Nit: This could also test the too short case

Reply:
it is? on line 475 we give it a string of length zero

Reply:
oh sorry... I overlooked that completely